core/src/main/java/org/apache/james/mime4j/util/MimeParameterMapping.java

```diff
@@ -0,0 +1,67 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mime4j.util;
+
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MimeParameterMapping {
+
+    private final Map<String, String> parameters = new HashMap<>();
+
+    public Map<String, String> getParameters() {
+        Map<String,String> result = new HashMap<>();
+        for (Map.Entry<String, String > entry : parameters.entrySet()) {
+            result.put(entry.getKey(), decodeParameterValue(entry.getValue()) );
+        }
+        return result;
+    }
+
+    public void addParameter(String name, String value) {
+        String key = removeSectionFromName(name).toLowerCase();
+        if (parameters.containsKey(key)) {
+            parameters.put(key, parameters.get(key) + value);
+        } else {
+            parameters.put(key, value);
+        }
+    }
+
+    private String decodeParameterValue(String value) {
+        int charsetEnd = value.indexOf("'");
```

Comment:
```suggestion
        if (value == null) {
            return null;
        }
        int charsetEnd = value.indexOf("'");
```

We likely should not try to decode null values returned by NameValuePair

Reply:
definitely, we don't want to decode null values, thanks for pointing it out